libs/content/workspace/src/workspace.rs

```diff
@@ -679,13 +691,16 @@ impl Workspace {
                     if let Some((name, id)) =
                         self.current_tab().map(|tab| (tab.name.clone(), tab.id))
                     {
-                        self.ctx.send_viewport_cmd(ViewportCommand::Title(name));
```

Comment:
this froze bc `send_viewport_cmd` ultimately tries to lock the same lock as `input_mut`